src/docs/lib/shiki.ts

```diff
@@ -11,35 +11,46 @@ const theme: Theme = "github-dark";
 const defaultlangs: Lang[] = ["html", "tsx"];
 const bg: React.CSSProperties["backgroundColor"] = "#011627";
 
+const highlighterCache = new Map<string, Promise<Highlighter>>();
+
+/**
+ * Get a shiki highlighter instance.
+ * Cached by theme and languages.
+ */
 export async function getHighlighter({
   langs = defaultlangs,
 }: {
   langs?: Lang[];
-} = {}) {
-  /** Preload NO languages in development */
-  const isDevelopment = process.env.NODE_ENV === "development";
+} = {}): Promise<Highlighter> {
+  const key = [theme, ...langs].join(",");
```

Comment:
Any reason for using a comma? Why not:

```suggestion
  const key = [theme, ...langs].join("-");
```